### ISSUE TITLE
In Compatibility with node 13.5

### ISSUE THREAD

**Issue by tamil777selvan** [webdriverio/webdriverio#4913]
**Describe your question with as much detail as possible**
Unable to run tests with node 13 as `wdio-sync` fibers are not getting loaded, got below log while execution:
```
2019-12-24T01:32:09.621Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
2019-12-24T01:32:09.624Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8
```

Tried uninstalling and installing all packages with latest version but no luck.

**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.18.0
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 13.5.0
 - **NPM version:** 6.13.4
 - **Browser name and version:** Chrome 79
 - **Platform name and version:** MAC
 - **Additional wdio packages used (if applicable):** 
```
     "@wdio/cli": "5.18.1",
    "@wdio/cucumber-framework": "5.16.15",
    "@wdio/local-runner": "5.18.0",
    "@wdio/reporter": "5.15.2",
    "@wdio/spec-reporter": "5.16.11",
    "@wdio/sync": "5.17.0",
```

**Comment by mgrybyk** [comment]
Have you deleted package lock before installing packages? Make sure node fibers of version 4.0.2 is installed https://github.com/laverdet/node-fibers/issues/424

**Comment by tamil777selvan** [comment]
Yea, I have deleted package lock file and installed all packages but not working. Also the fiber version is 4.0.2

Attached log file for reference where promise where not returned as expected and result in timeout error.
[wdio.log](https://github.com/webdriverio/webdriverio/files/4011763/wdio.log)

**Comment by christian-bromann** [comment]
I have to close this as I was not able to reproduce on:

- MacOS 10.14
- Node 13.5

I did the following:

```sh
$ cd /tmp
$ mkdir wdiotest
$ cd wdiotest
$ npm init --yes
$ npm i @wdio/cli
$ npx wdio config --yes
$ mkdir test
$ mkdir test/specs
$ vim test/specs/test.js
# wrote a simple mocha test
$ npx wdio
```

**Comment by snoblenet** [comment]
I am facing the same issue. WDIO tests run locally on Mac OS X Catalina but fail on Buildkite with error:

```
to be executed: node_modules/.bin/wdio webdriverio/wdio.conf.js
  | 2020-02-02T22:49:35.653Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
  | 2020-02-02T22:49:35.656Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8
  | 2020-02-02T22:49:35.819Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
  | 2020-02-02T22:49:35.820Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8
  | 2020-02-02T22:49:35.932Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
  | 2020-02-02T22:49:35.933Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8
  | 2020-02-02T22:49:35.952Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
  | 2020-02-02T22:49:35.952Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8

```

**Comment by snoblenet** [comment]
It would be extremely helpful if there as a ...
`log.error(e)`
...after the...
`log.debug('Couldn\'t load fibers package for Node v10 and above')`

**Comment by snoblenet** [comment]
Adding that to my own code revealed the underlying issue with fibres. Can I ask the the try/catch block does not completely swallow the error message?

**Comment by christian-bromann** [comment]
What is the error message?

**Comment by snoblenet** [comment]
Don't have the error in front of me, but it was one I was able to fix like this:

![image](https://user-images.githubusercontent.com/62277/73686392-fba7f900-471b-11ea-87b9-3439f9755398.png)

**Comment by alexneo2003** [comment]
I had the same problem

`2020-02-28T12:08:54.032Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
2020-02-28T12:08:54.036Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8`

and after `yarn install --force` the problem was fixed

**Comment by saranyaeaswaran** [comment]
Hi @snoblenet and @alexneo2003 , I'm facing the same issue and mine is OS 10.16 with npm, can you please suggest how can I fix this issue?

**Comment by alexneo2003** [comment]
@saranyaeaswaran maybe you can try remove node_modules and reinstall it

**Comment by christian-bromann** [comment]
Also, ensure that you have Python installed.

**Comment by AutomationGuru** [comment]
I had the same issue once the OS upgrade on mac, so i am listing what i did which helped me, 
I had node v13.6.0 and when i ran the test runs with `wdio <config>`

The following error was seen
```
 DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
DEBUG @wdio/sync: Couldn't load fibers package for Node v8

```
1. Try to install `npm i fibers`
2. i see the following error
`gyp: No Xcode or CLT version detected!
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1`
3. You need to run the following:
xcode-select --print-path
sudo rm -r -f /Library/Developer/CommandLineTools
xcode-select --install

4. once the install is done, you can re run the npm install or install fibers and that should work

**Comment by christian-bromann** [comment]
Please note that `@wdio/sync` is not a required package and you can use WebdriverIO without it.

**Comment by Sergiu-B** [comment]
I encountered this issue in AWS CodeBuild (our build tool in CI) after we upgraded the project's NodeJS version from 10.15.3 to 12.18.3. *Tests still worked fine locally (MacOS Catalina).
This is not a WebdriverIO issue, but still posting the solution since I wasted a couple of hours searching far and wide, and hopefully will save some time for someone.
`node-fibers` package does not have binaries for all supported versions and platforms, so, like in my case, you might need to create them.
This is pretty straightforward: `cd` into the fibers module and run `node build`. This is a script that the `fibers` maintainer created exactly for this purpose. Builds the binary and moves it to the correct location.
I added this to a shell script: 
```
cd "node_modules/fibers"
node build
cd "../../"
```
More details here: https://github.com/laverdet/node-fibers/issues/230

**Comment by yetyman** [comment]
> Yea, I have deleted package lock file and installed all packages but not working. Also the fiber version is 4.0.2
> 
> Attached log file for reference where promise where not returned as expected and result in timeout error.
> [wdio.log](https://github.com/webdriverio/webdriverio/files/4011763/wdio.log)

finally. someone else has the same issue as me. I am on Windows 10, developing an Electron project, very basic, totally vanilla. When Spectron tries to get the chromedriver session it fails. this eventually led me down a path of harder debugging starting with wdio/sync being missing. so I installed it, now fibers can't be loaded.

**Comment by unickq** [comment]
Faced today on CircleCi image update - 

node v14.15.1
`2020-11-17T13:54:33.581Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above`

> Please note that @wdio/sync is not a required package and you can use WebdriverIO without it.

@christian-bromann does it mean @wdio/sync needs to be replaced with webdriverio package? Currently, I have only sync installed

**Comment by DenysBarkan** [comment]
```
"node -v" : "v14.15.4"
"@wdio/cli": "^6.6.6"
```
returns the exception :
`"DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above"`

instead of the working flow

```
"node -v" : "v12.20.1"
"@wdio/cli": "^6.6.6"
```
works without any additional messages.

The downgrade of the node to v12 solves the problem with the issue.

**Comment by esod** [comment]
As a counterpoint,

`"node -v" : "v14.15.4"
"@wdio/cli": "^6.12.1"`

returns the exception
`2021-02-16T19:25:03.124Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above
wdio [command]`

install fibers
`npm i fibers --save-dev`

And all is well with `npx wdio`. `"fibers": "^5.0.0",` is the version that was installed.

**Comment by Sathish-elisetty** [comment]
> I had the same problem
> 
> `2020-02-28T12:08:54.032Z DEBUG @wdio/sync: Couldn't load fibers package for Node v10 and above 2020-02-28T12:08:54.036Z DEBUG @wdio/sync: Couldn't load fibers package for Node v8`
> 
> and after `yarn install --force` the problem was fixed

**Comment by tiecongli** [comment]
> Please note that `@wdio/sync` is not a required package and you can use WebdriverIO without it.

For people who need to run on sync mode requires this package tho

**Comment by christian-bromann** [comment]
@tiecongli we recommend to switch to async code execution and documented the process for this migration [here](https://webdriver.io/docs/async-migration). WebdriverIO will not support sync with the next major version anymore.

**Comment by kajmagnus** [comment]
@DenysBarkan and others — only this: `"fibers": "^5.0.0"` didn't work for me, because Webdriverio v6 depends on Fibers v4. If you just add Fibers v5, it'll get installed, but Webdriverio continues using v4. (As far as I've understoold, looking at `yarn.lock`.)

But by allso adding a `"resolutions": ...` entry to force Webdriverio to use Fibers v5, `sync` started working for me with Nodejs 14.

Here's my `package.json`:

```
{
  ...
  "devDependencies": {
    "@wdio/cli": "=6.10.10",
    ...
    "fibers": "v4.x doesn't work with Nodejs 14, which Ty uses — v12 is EOL (and odd versions like v13 are unstable).",
    "fibers": "5.0.1",
    ...
  },
  "resolutions": {
    "fibers": "Webdriver.io 6 tries to use Fibers 4.x.",
    "fibers": "5.0.1"
  }
}
```

@christian-bromann 

> we recommend to switch to async

Yes :- ) I've started, however, still about 180 Webdriverio v6 tests left to migrate to v7. About 1 whole weeks of work. It's nice that now I can do this a little bit at a time (since I got v6 working with Nodejs 14) instead of all at once.  Webdriver is really nice b.t.w. the best e2e test framework I've used.
